src/components/fields/location-input-group/location-modal/location-search/location-search.tsx

```diff
@@ -0,0 +1,749 @@
+			>
+				<ResultItemPin src={LocationPinBlack} alt="Location" />
+				<Text.BodySmall dangerouslySetInnerHTML={cleanHtml(item.displayText)}></Text.BodySmall>
+			</ResultItem>
+		));
+	};
+
+	const renderPostalCodeError = () => (
+		<Prompt
+			id={`${id}-postal-code-error`}
+			title="Oops"
+			size="large"
+			show={true}
+			description="The location you have selected does not contain a postal code. Please pick the building where the high-rise littering took place."
+			buttons={[
+				{
+					id: "ok",
+					title: "OK",
+					onClick: () => setShowPostalCodeError(false),
+				},
+			]}
+		/>
+	);
+
+	return (
+		<>
+			<SearchWrapper singelPanelInputMode={singelPanelInputMode}>
+				<SearchBarIconButton
+					onClick={handleClickCancel}
+					id={TestHelper.generateId("location-search-modal-close-icon")}
+				>
+					<SearchBarModalCross />
+				</SearchBarIconButton>
+
+				<SearchBarContainer hasScrolled={hasScrolled}>
+					<SearchBarIconButton
+						onClick={handleInputFocus}
+						id={TestHelper.generateId("location-search-modal-search-icon")}
+					>
+						<SearchBarIcon src={SearchSvg} alt="Search" />
+					</SearchBarIconButton>
+					<SearchBarInput
+						type="text"
+						onFocus={handleInputFocus}
+						onChange={handleInputChange}
+						placeholder={addressFieldPlaceholder}
+						readOnly={gettingCurrentLocation}
+						value={!gettingCurrentLocation ? queryString : gettingCurrentLocationFetchMessage}
+						ref={inputRef}
+						id={TestHelper.generateId("location-search-modal-search-text-input-field")}
+					/>
+
+					<SearchBarIconButton
+						onClick={handleClearInput}
+						id={TestHelper.generateId("location-search-modal-clear-icon")}
+					>
+						<SearchBarCross type="cross" />
+					</SearchBarIconButton>
+				</SearchBarContainer>
+				<ResultWrapper
+					singelPanelInputMode={singelPanelInputMode}
+					ref={resultRef}
+					onScroll={handleScrollResult}
+				>
+					{!gettingCurrentLocation && (
+						<>
+							{searchBuildingResults.length ? <ResultTitle>{locationListTitle}</ResultTitle> : null}
+							<InfiniteScrollList
+								items={renderList()}
+								loading={loading}
+								hasNextPage={hasNextPage}
+								loadMore={getMoreAddress}
+							/>
+							{!loading && useQuerySearch && searchBuildingResults.length === 0 && (
+								<NoResultTitle>No results found for &ldquo;{queryString}&rdquo;</NoResultTitle>
+							)}
+						</>
+					)}
+				</ResultWrapper>
+
+				<ButtonWrapper singelPanelInputMode={singelPanelInputMode}>
+					<ButtonItem buttonType="cancel" styleType="light" onClick={handleClickCancel}>
+						Cancel
+					</ButtonItem>
+					<ButtonItem
+						buttonType="confirm"
+						onClick={onConfirm}
+						disabled={
+							!queryString || selectedIndex < 0 || !selectedAddressInfo?.lat || !selectedAddressInfo?.lng
+						}
+					>
+						{isSinglePanel ? "Confirm location" : "Confirm"}
+					</ButtonItem>
+				</ButtonWrapper>
+			</SearchWrapper>
```

Comment:
- add `$` for transient props
- `TestHelper.generateId()` should reference the `id` passed to it, e.g. `TestHelper.generateId(id, "close-icon")` instead of `TestHelper.generateId(id, "location-search-modal-close-icon")`, this prevents id collision if there are multiple location fields